trecs/models/mf.py

```diff
@@ -0,0 +1,293 @@
+            is only kept for measurements and the system is unaware of it.
+
+        verbose: bool (optional, default: False)
+            If True, enables verbose mode. Disabled by default.
+
+        num_items_per_iter: int (optional, default: 10)
+            Number of items presented to the user per iteration.
+
+        model_params: dict (optional, default: None)
+            Arguments that can be passed to `lenskit.algorithms.als.ImplicitMF()` that
+            dictate model training. For example: `{'iterations':40, 'reg':0.01}`.
+
+        seed: int, None (optional, default: None)
+            Seed for random generator.
+
+    Attributes
+    -----------
+        Inherited by BaseRecommender: :class:`~models.recommender.BaseRecommender`
+
+    Examples
+    ---------
+        ImplicitMF can be instantiated with no arguments -- in which case,
+        it will be initialized with the default parameters and the item/user
+        representations will be assigned randomly.
+
+        >>> mf = ImplicitMF()
+        >>> mf.users_hat.shape
+        (100, 10)   # <-- 100 users (default), 10 latent features (default)
+        >>> mf.items.shape
+        (10, 1250) # <-- 10 latent features (default), 1250 items (default)
+
+        This class can be customized either by defining the number of users/items/latent
+        features in the system.
+
+        >>> mf = ImplicitMF(num_users=1200, num_items=5000)
+        >>> mf.users_hat.shape
+        (1200, 10) # <-- 1200 users, 10 attributes
+
+        >>> mf = ImplicitMF(num_users=1200, num_items=5000, num_latent_features=2000)
+        >>> mf.users_hat.shape
+        (1200, 2000) # <-- 1200 users, 2000 attributes
+
+        Note that all arguments passed in at initialization must be consistent -
+        otherwise, an error is thrown. For example, one cannot pass in
+        `num_users=200` but have `user_representation.shape` be `(300, 100)`.
+        Likewise, one cannot pass in `num_items=1000` but have
+        `item_representation.shape` be `(100, 500)`.
+
+        To use the model, it's advised to use the `startup_and_train` method, which
+        will collect user interaction data from random recommendations. At the end
+        of the startup period, an implicit matrix factorization model is fit to the
+        interaction data, generating new user/item latent representations. These
+        representations are fixed and used for future calls to `run`.
+
+        >>> mf = ImplicitMF(num_latent_features=10) # 10 latent features
+        >>> mf.startup_and_train(50) # 50 timesteps of training
+        >>> mf.run(100)
+
+        If you'd like to retrain the matrix factorization model at any point, you
+        may use the `fit_mf` method. It will fit a new set of latent user/item
+        representations based on the interaction data from the *most recent* call to
+        `run()`.
+
+        >>> mf.run(50) # data from this will not be used in fitting a new MF model
```

Comment:
I find this example confusing since the command is the same, and the comment specifies that different things will happen.

Reply:
I think I'll alter the comment to be a bit more understandable - what I was trying to say was that only interaction data from the most recent call to `run` will be used to fit a new model in a subsequent call to `fit_mf()`.